### ISSUE TITLE
`Point` does not handle `double.nan` values

### ISSUE THREAD

**Issue by bernardobelchior** [entronad/graphic#68]
When a Point whose coordinates contain `double.nan` is rendered, the following exception is thrown:

```
[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: 'dart:ui/painting.dart': Failed assertion: line 31 pos 10: '<optimized out>': Rect argument contained a NaN value.
#0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
#2      _rectIsValid (dart:ui/painting.dart:31:10)
#3      Path.addOval (dart:ui/painting.dart:2464:12)
#4      CircleShape.path (package:graphic/src/shape/point.dart:112:9)
#5      PointShapeBase.renderItem (package:graphic/src/shape/point.dart:69:23)
#6      PointShape.renderGroup (package:graphic/src/shape/point.dart:28:29)
#7      ElementRenderOp.render (package:graphic/src/geom/element.dart:235:46)
```

[In this draft PR](https://github.com/entronad/graphic/pull/65/files), there is a proposed solution. Let me know if you think that it is a good one, and I can make a final PR with tests, or if you want to take it and fix it 👍

**Comment by entronad** [comment]
Thanks, the NaN in PointShape should be handled.

There was such a step, but it seems I put the wrong `continue` instead of `return` so it dosen't work:
https://github.com/entronad/graphic/blob/047be27a79af442fd149097aae3147bec4b78da8/lib/src/shape/point.dart#L62

If you make a new PR I would be glad to merge. The only thing you need is to fix `continue` to `return` and add x assertion:
```
if (!point.dy.isFinite || !point.dy.isFinite) {
  return [];
}
```

**Comment by bernardobelchior** [comment]
Sure, here it is: https://github.com/entronad/graphic/pull/70

**Comment by entronad** [comment]
Thanks, it's perfect. (I didn't know the feature of Offset before :) )

**Comment by bernardobelchior** [comment]
Thank you 😄

**Comment by canastro** [comment]
@entronad is it possible to create a release with this fix?

**Comment by entronad** [comment]
Published in v0.9.2
